// @mainframe/transport-http-browser/esm/index.js
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inheritsLoose from "@babel/runtime/helpers/esm/inheritsLoose";
import _wrapNativeSuper from "@babel/runtime/helpers/esm/wrapNativeSuper";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var HEADERS = {
  Accept: 'application/json',
  'Content-Type': 'application/json'
};
var METHOD = 'POST';
export var HTTPError =
/*#__PURE__*/
function (_Error) {
  _inheritsLoose(HTTPError, _Error);

  function HTTPError(status, message) {
    var _this;

    _this = _Error.call(this, message) || this;

    _defineProperty(_assertThisInitialized(_this), "status", void 0);

    _this.status = status;
    return _this;
  }

  return HTTPError;
}(_wrapNativeSuper(Error));
export function resOrError(res) {
  return res.ok ? Promise.resolve(res) : Promise.reject(new HTTPError(res.status, res.statusText));
}
export default function createTransport(url) {
  return function request(data) {
    var request = {
      body: JSON.stringify(data),
      headers: HEADERS,
      method: METHOD
    };
    return fetch(url, request).then(resOrError).then(function (res) {
      return res.json();
    });
  };
}
