// @mainframe/transport-http-browser/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.resOrError = resOrError;
exports.default = createTransport;
exports.HTTPError = void 0;

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var _wrapNativeSuper2 = _interopRequireDefault(require("@babel/runtime/helpers/wrapNativeSuper"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var HEADERS = {
  Accept: 'application/json',
  'Content-Type': 'application/json'
};
var METHOD = 'POST';

var HTTPError =
/*#__PURE__*/
function (_Error) {
  (0, _inheritsLoose2.default)(HTTPError, _Error);

  function HTTPError(status, message) {
    var _this;

    _this = _Error.call(this, message) || this;
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "status", void 0);
    _this.status = status;
    return _this;
  }

  return HTTPError;
}((0, _wrapNativeSuper2.default)(Error));

exports.HTTPError = HTTPError;

function resOrError(res) {
  return res.ok ? Promise.resolve(res) : Promise.reject(new HTTPError(res.status, res.statusText));
}

function createTransport(url) {
  return function request(data) {
    var request = {
      body: JSON.stringify(data),
      headers: HEADERS,
      method: METHOD
    };
    return fetch(url, request).then(resOrError).then(function (res) {
      return res.json();
    });
  };
}
